### PR TITLE
added code for PUT"/campaigns/{campID}"

### PR DESCRIPTION
**Pull request by elnamy**

### CODE REVIEW

**Comment by sjfoz1019** [comment]
Doesn't compile because your other two commits pushed to main. I'll pull them into this branch then review.

File: adbackend/src/main/java/com/csc/adbackend/RestApi.java [hunk 1]
```diff
@@ -93,8 +93,16 @@
      * @return url of created campaign in Location header
      */
     @PutMapping(path = "/campaigns/{campID}")
-    public void updateCampInfo(@PathVariable Integer campID) {
-        // TODO
+    public ResponseEntity<String> updateCampInfo(@RequestBody Campaign campaign, @PathVariable Integer campID) {
```
Comment (sjfoz1019) [inline adbackend/src/main/java/com/csc/adbackend/RestApi.java:96]: Needs correct error handling

**Review by sjfoz1019 — CHANGES_REQUESTED** [review]
Doesn't show up as a file for me to review since it was pushed to main, but in CampaignServiceImpl.java:
```
@Override
    public void updateCampaign(Integer campId, Campaign campaign) {
        //campaign.setID(campId);
        campaignRepo.save(campaign);
    }
```
- Throws `Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is org.springframework.orm.jpa.JpaSystemException: ids for this class must be manually assigned before calling save(): com.csc.adbackend.Campaign; nested 
exception is org.hibernate.id.IdentifierGenerationException: ids for this class must be manually assigned before calling save(): com.csc.adbackend.Campaign] with root cause` because it's trying to create a new campaign from the input
- Should get existing campaign from Repo and update fields, then save

**Review by elnamy — COMMENTED** [review]
can someone check if the updateCampAd method works properly

**Pull request closed without merging**